packages/utils/node-resolver-rs/src/invalidations.rs

```diff
@@ -17,21 +17,25 @@ pub enum FileCreateInvalidation {
 
 #[derive(Default, Debug)]
 pub struct Invalidations {
-  pub invalidate_on_file_create: DashSet<FileCreateInvalidation>,
-  pub invalidate_on_file_change: DashSet<PathBuf>,
+  pub invalidate_on_file_create: RwLock<HashSet<FileCreateInvalidation>>,
+  pub invalidate_on_file_change: RwLock<HashSet<PathBuf>>,
```

Comment:
DashSet should be fine. This seems unrelated to the cache change. Putting the entire hashset behind a lock will be much slower.

Reply:
Hey @devongovett, from what I've measured, this isn't slower ; we won't accept any significant regression.

I'm keen to remove a few of the bits here if possible. `DashSet` should be fine if we trust it's implementation.

Reply:
What makes you think this is causing issues? Seems unrelated the potential memory safety problems you mentioned.

Dashmap is a super popular crate. It just shards the map into multiple disjoint maps under individual locks that can be accessed concurrently. I've previously measured improvements on benchmarks by using it. The resolver is quite hot and I did a lot of performance tuning on it.

Reply:
I was hoping to reduce the surface area of what can break.

As a follow-up or in this diff, I can include more benchmarks that try to replicate the performance gains that you've found. We can commit them onto the criterion suite and I can run them on both macOS and Linux/EC2.